app/controllers/users_controller.rb

```diff
@@ -30,6 +30,7 @@ def index
     else
       @users = User.joins(:role).where.not(id: User.first.id) #Remove first user
     end
+    @users = User.search(@users, params[:search]) if params[:search]
```

Comment:
I fixed an issue of not filtering by badge when searching, but other than that, I feel like everything is fine. Using `@users` is a perfectly okay thing to do, as you can just use a pre-existing variable instead of running a new query such as `User.all`. I may not understand your comment, however, and if so, please rephrase it. 😛